packages/flutter/lib/src/material/magnifier.dart

```diff
@@ -0,0 +1,330 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+
+/// {@template widgets.material.magnifier.magnifier}
+/// A [Magnifier] positioned by rules dictated by the native Android magnifier.
+/// {@endtemplate}
+///
+/// {@template widgets.material.magnifier.positionRules}
+/// Positions itself based on [magnifierSelectionOverlayInfoBearer]. Specifically, follows the
+/// following rules:
+/// - Tracks the gesture's X, but clamped to the beginning and end of the currently editing line.
+/// - Focal point may never contain anything out of bounds.
+/// - Never goes out of bounds vertically; offset until the entire magnifier is in the screen. The
+///   focal point, regardless of this transformation, always points to the touch Y.
+/// - If just jumped between lines (prevY != currentY) then animate for duration
+/// [jumpBetweenLinesAnimationDuration].
+/// {@endtemplate}
+class TextEditingMagnifier extends StatefulWidget {
```

Comment:
I wonder if `TextMagnifier` would be a better name since this is used non editable text as well.

Reply:
This is a good call, I'll rename and redo some comments that mislead to believe that this is only for editing. Thanks!